### ISSUE TITLE
PATCH: Eliminate incorrectly parsed Description lines from saved files

### ISSUE THREAD

**Issue by rsbx** [leozide/leocad#568]
This fixes the cause of the doubled '0 Name:' lines in saved files.

[0001-Eliminate-incorrectly-parsed-Description-lines-from-.patch.txt](https://github.com/leozide/leocad/files/5732002/0001-Eliminate-incorrectly-parsed-Description-lines-from-.patch.txt)

**Comment by leozide** [comment]
What case are you trying to fix with this change? I'm not sure it's correct, the spec states: "If the first line of a file is a line type 0 the remainder of the line is considered the file title (see Library Header specification). This overrides any META commands on that line."

So we should always save the first line with a comment even if the description is empty, otherwise the next line could be taken as the description when the file is loaded again.

**Comment by rsbx** [comment]
I have a number of mpd files where each individual ldr file in them has 2 'Name:' lines that don't always match. The likely cause was some tool eliminated the empty description line at the top of the ldr file and leocad (or lpub3d) came along later and took the 'Name:' line as the description and added the missing 'Name:' line back. The 2 'Name:' lines no longer match if I renamed the submodel.

Having gone back and carefully read the LDraw specs, I agree that leocad is conforming to the spec. I also believe that the spec is fundamentally flawed in this area.

- Removing the description empty test I added to the writing procedure is clearly correct.
- But keeping the preference for 'Name:' or 'Author:' tagged lines in the reading procedure is worth keeping, if non-conforming, because it's less likely to result in files that will mislead/confuse humans. And I think an intentional description starting with 'Name:' or 'Author:' is unlikely.
